src/Symfony/Component/ImportMaps/ImportMapManager.php

```diff
@@ -0,0 +1,131 @@
+<?php
+
+/*
+ * This file is part of the Symfony package.
+ *
+ * (c) Fabien Potencier <fabien@symfony.com>
+ *
+ * For the full copyright and license information, please view the LICENSE
+ * file that was distributed with this source code.
+ */
+
+namespace Symfony\Component\ImportMaps;
+
+use Symfony\Component\Filesystem\Filesystem;
+use Symfony\Component\HttpClient\HttpClient;
+use Symfony\Component\VarExporter\VarExporter;
+use Symfony\Contracts\HttpClient\HttpClientInterface;
+
+/**
+ * @author Kévin Dunglas <kevin@dunglas.dev>
+ */
+final class ImportMapManager
+{
+    /**
+     * @see https://regex101.com/r/2cR9Rh/1
+     *
+     * Partially based on https://github.com/dword-design/package-name-regex
+     */
+    private const PACKAGE_PATTERN = '/^(?:https?:\/\/[\w\.-]+\/)?(?:(?<registry>\w+):)?(?<package>(?:@[a-z0-9-~][a-z0-9-._~]*\/)?[a-z0-9-~][a-z0-9-._~]*)(?:@(?<version>[\w\._-]+))?(?:(?<subpath>\/.*))?$/';
+
+    private ?array $importMap = null;
+
+    public function __construct(
+        private readonly string $path,
+        private ?HttpClientInterface $httpClient = null,
+        private readonly Filesystem $filesystem = new Filesystem(),
+    ) {
+        $this->httpClient ??= HttpClient::createForBaseUri('https://api.jspm.io');
```

Comment:
This base URI should be configurable, as [JSPM recommend](https://jspm.org/docs/api#generator) to users to host and run the generator

Reply:
Done but AFAIU the web API itself is not intended to be self-hosted. JSPM Generator is a local CLI tool written in JS that we don't want to rely on.

> This API is created as a convenience for runtimes and services that are unable to easily run JavaScript in their environments to generate the map themselves.

That's exactly our case.